src/RemoteInboxNotifications/TransformerService.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+use InvalidArgumentException;
+use stdClass;
+
+/**
+ * A simple service class for the Transformer classes.
+ *
+ * Class TransformerService
+ *
+ * @package Automattic\WooCommerce\Admin\RemoteInboxNotifications
+ */
+class TransformerService {
+	/**
+	 * Create a transformer object by name.
+	 *
+	 * @param string $name name of the transformer.
+	 *
+	 * @return TransformerInterface|null
+	 */
+	public static function create_transformer( $name ) {
+		$camel_cased = lcfirst( str_replace( ' ', '', ucwords( str_replace( '_', ' ', $name ) ) ) );
+
+		$classname = __NAMESPACE__ . '\\Transformers\\' . $camel_cased;
+		if ( ! class_exists( $classname ) ) {
+			return null;
+		}
+
+		return new $classname();
+	}
+
+	/**
+	 * Apply transformers to the given value.
+	 *
+	 * @param mixed $target_value a value to transform.
+	 * @param array $transformer_configs transform configuration.
+	 *
+	 * @throws InvalidArgumentException Throws when one of the requried arguments is missing.
+	 * @return mixed|null
+	 */
+	public static function apply( $target_value, array $transformer_configs ) {
+		foreach ( $transformer_configs as $transformer_config ) {
+			if ( ! isset( $transformer_config->use ) ) {
```

Comment:
This could be done in `OptionRuleProcessor`'s `validate` function - any invalid transforms should invalidate the rule

Reply:
I think it's better if we catch exception from the `Option` class instead of validating the config in the `Option`. What do you think?

we could simply try and catch and invalidate the rule in case of any exception thrown from the `TransformerService`

Reply:
The idea of spec validation was to catch invalid specs before they got imported (during polling), rather than during execution. Plus, exceptions shouldn't be used for flow control.

Reply:
Looking at the flow of how the whole Inbox Remote Notification system works again, you're correct. I will make an update soon 👍